beacon-chain/rpc/validator/proposer.go

```diff
@@ -389,23 +397,20 @@ func (vs *Server) mockETH1DataVote(ctx context.Context, slot uint64) (*ethpb.Eth
 	}, nil
 }
 
-func (vs *Server) randomETH1DataVote(ctx context.Context) (*ethpb.Eth1Data, error) {
+func (vs *Server) randomETH1DataVote(ctx context.Context, beaconState *stateTrie.BeaconState) (*ethpb.Eth1Data, error) {
```

Comment:
please change this back too, we dont need to pass the state in here. Accessing the state for reading is cheap for the most part, we dont need to thread the state in.

Reply:
This is required for `inRangeVotes`, which is the problematic function that led to this PR.